packages/api-contract/src/Abi/index.ts

```diff
@@ -162,9 +162,18 @@ export class Abi {
   /**
    * Warning: Unstable API, bound to change
    */
-  public decodeEvent (data: Bytes | Uint8Array): DecodedEvent {
+  public decodeEvent (data: Bytes | Uint8Array, topic: EventRecord['topics'][0]): DecodedEvent {
+    // try to find a topic signature match - ink! v5 upwards
+    let event = this.events.find((e) => e.signatureTopic === topic.toHex());
```

Comment:
Not the prettiest but we would need support both way to identify the Event to be decoded, right?